packages/jest-matchers/src/matchers.js

```diff
@@ -645,11 +645,19 @@ const matchers: MatchersObject = {
       );
     }
 
-    const pass = equals(receivedObject, expectedObject, [
+    let pass = equals(receivedObject, expectedObject, [
       iterableEquality,
       subsetEquality,
     ]);
 
+    if (
+      receivedObject.constructor.name === 'Error' &&
+      expectedObject.constructor.name === 'Error' &&
+      receivedObject.message != expectedObject.message
+    ) {
+      pass = false;
+    }
```

Comment:
How about:
```js
  const pass = receivedObject.constructor.name === 'Error' &&
      expectedObject.constructor.name === 'Error' &&
      receivedObject.message !== expectedObject.message
      ? false
      : equals(receivedObject, expectedObject, [
          iterableEquality,
          subsetEquality,
        ]);
```
I'm not sure if it's more readable, but we don't have to reassign `pass` value.